### DISCUSSION TITLE
I want to delete image by reading its base64 value. I cannot find any suitable sample code. Do you have? steve8000818@gmail.com

### DISCUSSION THREAD

**Discussion by nissansz** [pymupdf/PyMuPDF#2024] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Potentially add an issue reference.
I want to delete image by reading its base64 value. I cannot find any suitable sample code. Do you have? steve8000818@gmail.com

**Describe the solution you'd like**

**Describe alternatives you've considered**
Are there several options for how your request could be met?

**Additional context**

**Comment by JorjMcKie** [comment]
This is a typical "Discussions" item and does not qualify as an issue.
So I will transfer it and answer from there.

**Reply by nissansz** [reply]
Ok. Will wait for answer.

**Comment by nissansz** [comment]
When I try to use remover.py, below errors occurred. Sometimes the code works.
I want to remove the small image under the large image by judging base64 value of the image.
Process finished with exit code -1073741819 (0xC0000005)
[59-60.pdf](https://github.com/pymupdf/PyMuPDF/files/9947023/59-60.pdf)

![image](https://user-images.githubusercontent.com/41010669/200199873-e2449ade-becf-4326-9f0b-9bddfe9f8377.png)

**Comment by nissansz** [comment]
After disable this line # page.clean_contents()  # unify page's /Contents into one
The code can run, but it removed other images too, below is result, don't know how to solve. 

![image](https://user-images.githubusercontent.com/41010669/200201683-b6afe69d-96be-4fec-833a-f0de52309c15.png)

**Comment by nissansz** [comment]
I tried image-maintenance.py below is error. 

Python: 3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)]
wxPython: 4.1.1 msw (phoenix) wxWidgets 3.1.5

PyMuPDF 1.20.2: Python bindings for the MuPDF 1.20.3 library.
Version date: 2022-08-13 00:00:01.
Built for Python 3.8 on win32 (64-bit).


Process finished with exit code -1073741819 (0xC0000005)

**Reply by JorjMcKie** [reply]
Please update your PyMuPDF to the latest v1.21.0.
It fixes a problem which sometimes cause segmentation faults when executing `page.clean_contents()`.
I tried your PDF with the GUI script and it did work.

**Comment by nissansz** [comment]
I cannot access the new package.
Any place to download for offline install?

**Reply by JorjMcKie** [reply]
Sorry, it is still a release **candidate**, so please download like [here](https://github.com/pymupdf/PyMuPDF/discussions/2009)

**Comment by nissansz** [comment]
python -m pip install pymupdf==1.21.0rc1 --trusted-host pypi.org pytest-html
This command works. I installed it.

There is a leftover transparent rect in the pdf, cannot remove too?

![image](https://user-images.githubusercontent.com/41010669/200256652-82a59141-dd64-467e-b1ca-2c7ea283e168.png)

**Reply by JorjMcKie** [reply]
which rect do you mean?

**Comment by nissansz** [comment, marked as answer]
The original image. Which seems removed, but there is a hidden rectangular, as you can see in above image.

**Reply by JorjMcKie** [reply]
If that is no image and no text, it cannot be removed.

**Comment by nissansz** [comment]
![image](https://user-images.githubusercontent.com/41010669/200258757-afa51c2d-bb15-484d-9a9b-981471702629.png)

Also, I want to replace some hidden white text with space, or just remove those white text. Any method or code?

**Reply by JorjMcKie** [reply]
Use redaction annotations. You need the boundary box of the unwanted text.
Then do:
```python
page.add_redact_annot(bbox)
# maybe add more redact annots
# finally:
page.apply_redactions()
```

**Comment by nissansz** [comment]
Is it possible to get html for redact instead of get_text?

    for block in page.get_text("dict", flags=fitz.TEXTFLAGS_TEXT)["blocks"]:

**Reply by JorjMcKie** [reply]
Any method by which you can determine the rectangle of unwanted text is okay.

**Reply by JorjMcKie** [reply]
(Redaction) annotations as such have no relation to text or other page content.
Annotations are not a part of the page content - compare them like dust on the glass cover of a picture: can be wiped away without changing the picture itself.

**Comment by nissansz** [comment]
Thank you. I will hold the attempt to redact text.
If there is such demand in future, will try again.
